src/charm.py

```diff
@@ -427,10 +427,11 @@ def _check_and_update_dependencies(self) -> bool:
             self.unit.status = MaintenanceStatus(f"Failed to check for runner updates: {err}")
             return False
 
+        # TODO: Remove debugging.
+        logger.info("%s %s", runner_info.download_url, self._stored.runner_bin_url)
```

Comment:
Would it be possible to leave debugging on and only trigger it conditionally, I recall bitbucket pipelines let's you define variables on runs, so you might pass debug=on and it defaults to debug=off for instance..

Reply:
This is converted to debug-level logging.